File: src/types/source/_Erratum.ts
/**
 * @main Errata
 */

import * as DB from "tsondb/schema/dsl"

export const Errata = DB.TypeAlias(import.meta.url, {
  name: "Errata",
  comment: "A list of errata for the entry in the specific language.",
  type: () => DB.Array(DB.IncludeIdentifier(Erratum), { minItems: 1 }),
})

export const Erratum = DB.TypeAlias(import.meta.url, {
  name: "Erratum",
  type: () =>
    DB.Object({
      date: DB.Required({
        comment: "The date when the change was confirmed and applied to the entry.",
        type: DB.Date(),
      }),
      description: DB.Required({
        comment: "A description of what changed.",
        type: DB.String({ minLength: 1 }),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/source/_Erratum.swift
//
//  _Erratum.swift
//  OptolithDatabaseSchema
//

/// A list of errata for the entry in the specific language.
public typealias Errata = [Erratum]

public struct Erratum: EntitySubtype {
    /// The date when the change was confirmed and applied to the entry.
    public let date: String

    /// A description of what changed.
    public let description: NonEmptyMarkdown

    public init(date: String, description: NonEmptyMarkdown) {
        self.date = date
        self.description = description
    }
}
